packages/runtime/src/next/api-handler/handlers/proxy-draft-mode.ts

```diff
@@ -59,11 +59,16 @@ async function proxyDraftModeRouteHandler(
   const draft = await draftMode()
   draft.enable()
 
-  const proxyUrl = request.nextUrl.clone()
-  proxyUrl.searchParams.delete('x-makeswift-draft-mode')
+  const forwardingHost = request.headers.get('x-forwarded-host') ?? request.headers.get('host') 
+  const forwardingProto = request.headers.get('x-forwarded-proto') ?? request.nextUrl.protocol
 
-  const proxyRequest = new NextRequest(proxyUrl, { headers: request.headers })
-  proxyRequest.headers.delete('x-makeswift-draft-mode')
+  const searchParams = new URLSearchParams(request.nextUrl.searchParams)
+  searchParams.delete('x-makeswift-draft-mode')
+  const headers = new Headers(request.headers)
+  headers.delete('X-Makeswift-Draft-Mode')
+
+  const proxyUrl = `${forwardingProto}//${forwardingHost}${request.nextUrl.pathname}?${searchParams}`
```

Comment:
Can't `forwardingHost` be `undefined` here? In my suggestion in my review (in the other PR) this is avoided by making sure we only override the host from the original URL, if present: https://github.com/makeswift/makeswift/pull/907#discussion_r1911759234